7/README.md

```diff
@@ -0,0 +1,47 @@
+```
+shortname: 7/PUBLIC-API
+name: Definition of the BigchainDB Public API
+type: Informational
+status: raw
+editor: Troy McConaghy <troy@bigchaindb.com>
+```
+
+# Abstract
+
+The "public interface" or "public API" of any software is the stuff that _external_ users or developers see and use to interact with it. These are different from internal APIs, which are used only by the developers developing the internal "black box" code. This BEP summarizes what's included (and not included) in the "BigchainDB Public API." It also explains what it means that BigchainDB follows semantic versioning (and which variety of that).
+
+# Motivation
+
+The definition of the BigchainDB Public API was in the `CHANGELOG.md` file, which is crazy because it's not obvioius that's the place to look for such a thing.
+
+External users want to be able to guess, _based on the version number of a release_, what has changed, if anything. To make that possible, BigchainDB Server (and affiliated projects) use [the Python form of Semantic Versioning](https://packaging.python.org/tutorials/distributing-packages/#choosing-a-versioning-scheme) (or at least we try). It's almost the same as regular semantic versioning except release candidates are labelled like `3.4.5rc2` not `3.4.5-rc2` (with no hyphen).
+
+# Specification
+
+The _BigchainDB Server_ **Public API** includes:
+
+* The configuration defaults, schemas, and paths where configuration files are stored.
+* The Command Line Interface, i.e. `bigchaindb subcommand STUFF`
+* The transaction spec. Currently known as the IPDB Transaction Spec, but that may get renamed.
+* The HTTP API (including payloads sent to and received from them, so it includes transactions here again).
+* The Events API, also called the Event Stream API.
+
+The BigchainDB Public API excludes:
+
+* Everything marked as "experimental" (even if the feature is part of the Public API).
+* The logs generated by the server.
+* The internal API, such as classes, methods, functions.
+
+# Documentation of the Current BigchainDB Public API
+
+At the time of writing, the various components of the BigchainDB Public API were documented in various places. Those were:
+
+* [Configuration settings (latest release)](https://docs.bigchaindb.com/projects/server/en/latest/server-reference/configuration.html)
+* [Command Line Interface (latest release)](https://docs.bigchaindb.com/projects/server/en/latest/server-reference/bigchaindb-cli.html)
+* [Transaction Spec (all versions)](https://github.com/ipdb/ipdb-tx-spec)
+* [HTTP API (latest release)](https://docs.bigchaindb.com/projects/server/en/latest/http-client-server-api.html)
+* [Events API (latest release)](https://docs.bigchaindb.com/projects/server/en/latest/events/index.html)
```

Comment:
Do you think we should link to a specific version of the documentation at one point? (where *at one point* is when we have the documentation for BigchainDB 2.0 online).

Reply:
That is a good idea, but right now there are no URLs for BigchainDB 2.0, e.g. on ReadTheDocs, because it doesn't exist yet. It will come from a Git Tag on the commit which _defines_ BigchainDB 2.0, which will also be tied to the GitHub _release_. That Git Tag doesn't exist yet, but it will soon! And then we can make links to all the BigchainDB 2.0 specific specs. Yay!